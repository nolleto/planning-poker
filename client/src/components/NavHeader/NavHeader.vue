<template>
  <nav class="wrapper flex-center">
    <a
      class="title"
      @click="$emit('go-home')"
    >
      Planning Poker
    </a>

    <div class="flex-center">
      <div class="divider" />
      <nav-login />
    </div>
  </nav>
</template>

<script>
import NavLogin from '../NavLogin'

export default {
  name: 'NavHeader',

  components: { NavLogin }
}
</script>

<style scoped>
  .wrapper {
    padding: 10px 20px;
    background-color: #f4f4f4;
  }

  .flex-center {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .divider {
    height: 20px;
    margin-right: 10px;
    border-left: 1px solid #dddddd;
  }

  .title {
    padding: 5px 0;
    cursor: pointer;
    text-decoration: none;
    font-weight: 500;
  }
</style>
