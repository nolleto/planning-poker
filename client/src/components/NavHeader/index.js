export { default } from './NavHeader.vue'
