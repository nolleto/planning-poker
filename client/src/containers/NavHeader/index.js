import { connect } from 'vuex-connect'
import router from '../../routers'
import NavHeader from '../../components/NavHeader'

export default connect({
  actionsToEvents: {
    'go-home': () => router.push('/')
  }
})('NavHeaderContainer', NavHeader)
