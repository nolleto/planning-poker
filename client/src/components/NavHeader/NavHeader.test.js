import NavHeader from './index'
import { shallowMount } from '@vue/test-utils'

const shallowComponent = () => {
  return shallowMount(NavHeader)
}

describe('NavHeader component', () => {
  let wrapper

  describe('Snapshots', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    it('matches', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Event Listeners', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    describe('When clicked in home link', () => {
      beforeEach(() => {
        jest.spyOn(wrapper.vm, '$emit')
        wrapper.find('a').trigger('click')
      })

      it('emits "go-home"', () => {
        expect(wrapper.vm.$emit).toHaveBeenCalledWith('go-home')
      })
    })
  })
})
