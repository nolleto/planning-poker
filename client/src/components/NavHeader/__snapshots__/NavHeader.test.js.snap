// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavHeader component Snapshots matches 1`] = `
<nav
  class="wrapper flex-center"
>
  <a
    class="title"
  >
    
    Planning Poker
  
  </a>
   
  <div
    class="flex-center"
  >
    <div
      class="divider"
    />
     
    <navlogin-stub />
  </div>
</nav>
`;
